src/adagrams.js

```diff
@@ -1,15 +1,140 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+
+function getRandomProperty() {
+  const keys = Object.keys(LETTER_POOL);
+
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+let drawnLetters = [];
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  while (drawnLetters.length < 10) {
+    let myCurrentKey = getRandomProperty();
+    let myCurrentValue = LETTER_POOL[myCurrentKey];
+    if (myCurrentValue !== 0) {
+      drawnLetters.push(myCurrentKey);
+      LETTER_POOL[myCurrentKey] = myCurrentValue - 1;
+    }
+    if (LETTER_POOL[myCurrentKey] === 0) {
+      delete LETTER_POOL[myCurrentKey];
+    }
+  }
+  
+  return drawnLetters;
```

Comment:
This function looks good but I have two suggestions:

1) I would make a copy of LETTER_POOL because as it stands, you are destroying LETTER_POOL as you remove letters.
2) You don't ever account for the different numbers of letters, so the randomness accounts for the 26 different keys, but it weights them each equally. To get around this, we would want to create some sort of structure that holds all the different letters and randomly picks one from there!